src/LoggingFormats.jl

```diff
@@ -1,5 +1,39 @@
 module LoggingFormats
 
-greet() = print("Hello World!")
+using Logging
+
+function shorten_str(message, max_len)
+    suffix = "…"
+    if length(message) > max_len
+        message[1:min(end, max_len-length(suffix))] * suffix
```

Comment:
I don't think this works with Unicode
```
julia> shorten_str("αβγαβγ", 3)                                                                                    │
ERROR: StringIndexError: invalid index [2], valid nearby indices [1]=>'α', [3]=>'β'
```

I guess we have to decide whether it should be max number of characters, and in that case I think `textwidth` is what you want and not length? I am not sure how to truncate to `x` characters though other than iteration through the string.

Reply:
Here is one suggestion:
```julia
function shorten_str(str, max_len)
    io = IOBuffer(; sizehint=max_len)
    len = 0
    for i in eachindex(str)
        c = str[i]
        len += textwidth(c)
        if len > max_len
            break
        end
        print(io, c)
    end
    return String(take!(io))
end
```

Reply:
Thanks. I solved it using `nextind` which I think is a bit more elegant.

Reply:
But that won't respect `textwidth`, right? See https://julialang.zulipchat.com/#narrow/stream/274208-helpdesk-.28published.29/topic/Truncate.20string.20to.20X.20characters for some discussion.

Since this truncated string is just later printed we could also return `SubString(str, 1, ind)` from the function and avoid the extra string allocation.